tabular/src/autogluon/tabular/models/catboost/catboost_model.py

```diff
@@ -62,7 +62,8 @@ def _preprocess_nonadaptive(self, X, **kwargs):
     def _estimate_memory_usage(self, X, **kwargs):
         num_classes = self.num_classes if self.num_classes else 1  # self.num_classes could be None after initialization if it's a regression problem
         data_mem_usage = get_approximate_df_mem_usage(X).sum()
-        approx_mem_size_req = data_mem_usage * 7 + data_mem_usage / 4 * num_classes  # TODO: Extremely crude approximation, can be vastly improved
+        iterations = self.params.get('iterations', DEFAULT_ITERATIONS)
+        approx_mem_size_req = 76.55 * data_mem_usage + 25.24 * num_classes - 0.16 * iterations - 9.38 * len(self.features)  # Linear regression from collected memory profiling data points
```

Comment:
Because `data_mem_usage` is astronomically larger than all the other variables, it is the only thing that matters in this equation. `76.55` is a very large modifier. I'm not confident that this is going to be correct.